### DISCUSSION TITLE
How to handle different monitor names

### DISCUSSION THREAD

**Discussion by kalsan** [i3/i3#5973] [Q&A] [answered]
I'm using i3 in a multi monitor setup using a USB-C dock. The problem is that my external monitor's name changes across boots / reconnects (e.g. yesterday it was named `DP-1-3-1`, today it shows up as `DP-3-3-1`.

While I can easily script screen autodetection with xrandr and bash for keyboard-run screen management scripts (e.g. extend right), I can't do that (I think) with the assignment of workspaces to screens. For instance, I use:

- `set $right_monitor DP-1-3-1`
- `workspace 1› output $right_monitor`

Since the screen's name keeps changing, this config breaks regularly.

Any ideas on how I can dynamically detect the current external screen for workspace assignment?

Best,
Kalsan

**Comment by KebabLord** [comment, marked as answer]
you can use "primary" and "nonprimary" keys after version 4.23+
```
# Assign workspace 1 to first monitor, workspace 2 to second monitor
workspace 1 output primary
workspace 2 output nonprimary
```

**Reply by kalsan** [reply]
Thanks a lot! Surprisingly this also appears to work under `i3 version 4.22 (2023-01-02)`
